akka-actor/src/main/scala-2.13+/akka/util/ByteString.scala

```diff
@@ -645,20 +645,19 @@ object ByteString {
     }
 
     override def copyToArray[B >: Byte](dest: Array[B], start: Int, len: Int): Int = {
-      if (isCompact) bytestrings.head.copyToArray(dest, start, len)
+      if (bytestrings.size == 1) bytestrings.head.copyToArray(dest, start, len)
```

Comment:
One small optimization added here, if there is just one sub-string we can always call copyToArray on that directly, not only when it is compact

Reply:
Has that also be done for the 2.12 version?

Reply:
Anyway, the concept of "compact ByteStrings" seems weird to me anyway. It would mean that there's useless nesting of `ByteStrings` around a single compact core. IMO that should probably be forbidden.

Reply:
I'm not sure how/if it will end up with that scenario ever to be honest. Just something I noticed.